### PR TITLE
Auth System Tests (#657)

### LINKED ISSUE #657: Auth system test
The current system tests use the default admin user for all requests.

The auth test should create users with less privileges and check actions are passing or blocked according to the permissions.

### PR DESCRIPTION
**Pull request by adunuthulan**
Fixes #657 

Created system tests using different clients, each with the policies of the user groups Developers, SuperUsers, and Viewers. Each test checks to see if the client correctly responds to the user's assigned policies. I plan on adding merging to the set of tests.

### CODE REVIEW

File: nessie/auth_test.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+		Name:   branch1,
+		Source: mainBranch,
+	})
+	require.NoError(t, err, "SuperUser failed while testing create branch")
+	require.Equal(t, http.StatusCreated, resAddBch.StatusCode(), "SuperUser unexpectedly failed to create branch")
+
+	// attempting to delete the repository should be authorized and result in no content response
+	resDelRep, err := superUserClient.DeleteRepositoryWithResponse(ctx, repo)
+	require.NoError(t, err, "SuperUser failed while testing delete repository")
+	require.Equal(t, http.StatusNoContent, resDelRep.StatusCode(), "SuperUser unexpectedly did not receive no content response while deleting repo")
```
Comment (itaiad200) [inline nessie/auth_test.go:90]: The admin can do so much more, like managing groups. Let's add more checks for those as well.

**Review by itaiad200 — CHANGES_REQUESTED** [review]
Really enjoyed reviewing this code, it is clean, concise and well-documented. The PR description is also very helpful in understanding where this stands.

Feel free to undraft for next iteration after fixing the comments and adding the tests that you wrote.

**Review by itaiad200 — APPROVED** [review]
Thanks for fixing the comments. Approved

File: pkg/testutil/setup.go [hunk 2]
```diff
@@ -58,15 +58,7 @@ func SetupTestingEnv(params *SetupTestingEnvParams) (logging.Logger, api.ClientW
...
-	endpointURL := viper.GetString("endpoint_url")
-	u, err := url.Parse(endpointURL)
-	if err != nil {
-		logger.WithError(err).Fatal("Failed to parse endpoint URL", endpointURL)
-	}
-
-	if u.Path == "" || u.Path == "/" {
-		endpointURL = strings.TrimRight(endpointURL, "/") + api.BaseURL
-	}
+	endpointURL := ParseEndpointURL(logger, viper.GetString("endpoint_url"))
```
Comment (arielshaqed) [inline pkg/testutil/setup.go:61]: Hurrah! :confetti_ball:

**Review by arielshaqed — COMMENTED** [review]
Thanks for testing a really important part of the system!  This is a really good test.

I am asking for some refactoring it to a more data-driven style.  That is the Go Way, but more importantly I believe that it has significant advantages here in making it easier to review and maintain the code.

# There is :snail:  *NO URGENCY* :snail: in doing this!  The repo is better with this test than without it!

If you like I'd be very happy to go over how we might improve maintainability here.  Let me know :-)

Thanks!

**Pull request merged**